Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -0,0 +1,258 @@
+                var balance = GetLargestBalance(peopleBalances);
+                SetNames(ref name, ref maxBalance, peopleBalances, balance);
+            }
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+
+            var maxBalanceNoComma = RemoveComma($"{maxBalance:C0}");
+            return $"{FixPlural(name.ToString())} had the most money ever. {maxBalanceNoComma}.";
+        }
+
+        private static decimal GetLargestBalance(string[] peopleBalances)
+        {
+            var maxBalance = 0m;
+
+            for (var i = 1; i < peopleBalances.Length; i++)
+            {
+                var isBalanceValid = decimal.TryParse(peopleBalances[i], out var balance);
+
+                if (isBalanceValid && balance > maxBalance)
+                {
+                    maxBalance = balance;
+                }
+            }
+
+            return maxBalance;
+        }
+
+        private static void SetNames(ref StringBuilder name, ref decimal maxBalance, string[] peopleBalances, decimal balance)
+        {
+            if (balance >= maxBalance)
+            {
+                if (balance > maxBalance)
+                {
+                    name = new StringBuilder();
+                }
+
+                if (name.Length != 0)
+                {
+                    name.Append(", ");
+                }
+
+                name.Append(peopleBalances[0]); //Builds list of names with highest balance
+                maxBalance = balance;
+            }
+        }
+
+        /// <summary>
+        /// Return name and loss of a person with a biggest loss (balance change negative).
+        /// </summary>
+        public string FindPersonWithBiggestLoss(string[] peopleAndBalances)
+        {
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return NotAvailable;
+            }
+
+            var biggestLossEver = decimal.MaxValue;
+            var name = "";
+
+            for (var i = 0; i < peopleAndBalances.Length; i++)
+            {
+                var peopleBalances = peopleAndBalances[i].Split(",");
+
+                if (peopleBalances.Length <= 2)
+                {
+                    continue;
+                }
+
+                var biggestLossForPerson = GetBiggestLossForPerson(peopleBalances);
+
+                if (biggestLossForPerson < biggestLossEver)
+                {
+                    biggestLossEver = biggestLossForPerson;
+                    name = peopleBalances[0];
+                }
+            }
+
+            if (name == "")
+            {
+                return NotAvailable;
+            }
+
+            CultureInfo.CurrentCulture = CultureInfo.InvariantCulture;
+
+            var biggestLossEverNoComma = RemoveComma($"{biggestLossEver:C0}").Replace("(", "-").Replace(")", "");
+            return $"{name} lost the most money. {biggestLossEverNoComma}.";
+        }
+
+        private decimal GetBiggestLossForPerson(string[] peopleBalances)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Make 'GetBiggestLossForPerson' a static method.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5063788)